### ISSUE TITLE
WindowManager$BadTokenException inside the AppUpgradeRepository.showForceUpgradePopup

### ISSUE THREAD

**Issue by vitaliygerasymchuk** [appupgrade-dev/app-upgrade-android-sdk#1]
Hello, we're using your SDK to handle the app upgrades. We are experiencing the crash: 

```
Fatal Exception: android.view.WindowManager$BadTokenException: Unable to add window -- token android.os.BinderProxy@ac2c11c is not valid; is your activity running?
       at android.view.ViewRootImpl.setView(ViewRootImpl.java:904)
       at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:342)
       at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:97)
       at android.app.Dialog.show(Dialog.java:419)
       at com.appupgrade.app_upgrade_android_sdk.AppUpgradeRepository.showForceUpgradePopup$lambda-3$lambda-2(AppUpgradeRepository.kt:110)
       at com.appupgrade.app_upgrade_android_sdk.AppUpgradeRepository.$r8$lambda$6C8omhxRfxmIwNHVKU2AEAHQBKI(AppUpgradeRepository.kt)
       at com.appupgrade.app_upgrade_android_sdk.AppUpgradeRepository$$ExternalSyntheticLambda1.run(D8$$SyntheticClass)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6682)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```
       
       
      

>   Having examined your code - i can see that asynchronous onResponse callback doesn't check if activity is alive neither is killing the api call when it's gonna be destroyed which causes the crash provided above. Could you please fix this?
>        Regards, Vitaliy

**Comment by appupgrade-dev** [comment]
Thanks, we will take a look.

**Comment by appupgrade-dev** [comment]
@vitaliygerasymchuk Hi, we have added fix for the issue and released a new version 1.0.3, Please update the SDK and try it now. If you face any issues let us know.

**Comment by vitaliygerasymchuk** [comment]
@appupgrade-dev thanks, will try it out shortly
